### PR TITLE
feat: impl Add, Sub, Mul, Div for all newtypes.

### LINKED ISSUE #277: Arithmetic on low-level types
The various new types could use implementations for several traits from
[std::ops](https://doc.rust-lang.org/std/ops/index.html#traits).

There's a good bit of boiler-plate to this, as we need symmetry for
all non-Self types involved in the ops. (We definitely want f64
supported.)

Because subtracting two, say, Time values can lead to a negative number,
it seems that the output of such ops should be f64.

### PR DESCRIPTION
**Pull request by alxsimon**
Closes #277

### CODE REVIEW

File: demes/tests/test_arithmetics.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+use demes::{CloningRate, DemeSize, GenerationTime, MigrationRate, Proportion, SelfingRate, Time};
+
+macro_rules! test_newtype_arithmetics {
+    ($type: ty, $fn_name: ident, $a:expr,$b:expr) => {
+        #[test]
+        fn $fn_name() {
+            let a = $a;
```
Comment (molpopgen) [inline demes/tests/test_arithmetics.rs:7]: Overall, this looks great -- thanks!

I have some concerns with the tests

1. For some of the types, 0.0 is not allowed.  For example, Time can be 0 but DemeSize cannot. That case is not covered here.
2. One way to stress test these operations in more depth is via [proptest](https://docs.rs/proptest/latest/proptest/).  We could add that as a dev dependency and then let it test all sorts of edge cases for the input into these ops.  The simplest method is to use their [macro](https://altsysrq.github.io/proptest-book/proptest/tutorial/macro-proptest.html) to do the heavy lifting.  The idea would be to let proptest generate two f64.  We can then say `let t = DemeSize::try_from(a + b)`.  If `t.is_ok()`, then we also expect DemeSize(a) + b, a + DemeSize(b), etc., to all be Some(...) and to contain the same value.

Comment (alxsimon) [reply demes/tests/test_arithmetics.rs:7]: Thanks, I'll have a look into proptest when I have time. Thant sounds like a good solution.

Comment (molpopgen) [reply demes/tests/test_arithmetics.rs:7]: There are [limitations](https://altsysrq.github.io/proptest-book/proptest/limitations.html) to property testing.  Upon a bit of reflection, It may be best now to manually treat edge cases: NaN input (f64::NAN), 0 for DemeSize, etc..

Comment (molpopgen) [reply demes/tests/test_arithmetics.rs:7]: @alxsimon -- I am happy to merge this and revisit detailed testing later.  Overall, these crates could be doing a better job testing the newtypes throughout.  I think it'll be more efficient to do a higher level review of the testing and make changes later.

Comment (alxsimon) [reply demes/tests/test_arithmetics.rs:7]: Feel free to merge this one, good for me to come back to tests later
